### PR TITLE
move enum protos into separate folder/package

### PR DESCRIPTION
**Pull request by MarlonGamez**
**Related**: #5368 

Apologies for large delta, most of this is generated files.

**Description**
This moves the enum definitions from proto/v1/skaffold.proto into proto/enums/enums.proto. This is to keep all of our enum definitions in one place with the addition of proto/v2, which will use the enums as well. To keep enum definitions on the docs page, I generate the enum docs, then the API docs and append the enum docs to the API ones. 

I've also updated some of the dependencies used to generate docs, as I think the updates to the dependencies make the docs a bit more readable by clarifying between enum types and standard proto message types.

### CODE REVIEW

**Comment by MarlonGamez** [comment]
Looks like this is causing issues with the HTTP docs generation, I'm looking into how I can fix this

**Review by tejal29 — DISMISSED** [review]

**Review by tejal29 — COMMENTED** [review]
LGTM!

**Comment by MarlonGamez** [comment]
@tejal29 I think the HTTP docs should be fixed now, PTAL when you get a chance :)

**Review by tejal29 — APPROVED** [review]

File: docs/content/en/docs/references/api/grpc.md [hunk 1]
```diff
@@ -19,11 +19,11 @@ We also generate the [reference doc for the HTTP layer]({{<relref "/docs/referen
 
 
 
-<a name="skaffold.proto"></a>
```
Comment (briandealwis) [inline docs/content/en/docs/references/api/grpc.md:22]: The comments at the top of this file references `proto/markdown.tmpl`.  Maybe `hack/generate_proto.sh` needs to be fixed up?

File: proto/v1/skaffold.pb.go [hunk 2]
```diff
@@ -1,11 +1,12 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
-// source: skaffold.proto
+// source: v1/skaffold.proto
 
 package proto
 
 import (
 	context "context"
 	fmt "fmt"
+	enums "github.com/GoogleContainerTools/skaffold/proto/enums"
```
Comment (briandealwis) [inline proto/v1/skaffold.pb.go:9]: Can this be sorted to preserve the Skaffold style of placing Skaffold imports at the end?

**Pull request merged**